golem_messages/validators.py

```diff
@@ -19,3 +19,133 @@ def validate_integer(field_name, value):
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_boolean(field_name, value):
+    if not isinstance(value, bool):
+        raise exceptions.FieldError(
+            f"{field_name} should be boolean",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_list(field_name, value, min_length, max_length):
+    if not (isinstance(value, list) and
+            (min_length <= len(value) <= max_length)):
+        raise exceptions.FieldError(
+            f"{field_name} should be a list with minimum {min_length}"
+            f" element and maximum {max_length} elements.",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_list_of_positive_integers(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if not is_number_positive(element, int):
+            raise exceptions.FieldError(
+                f"{field_name} should be a list with positive integers",
+                field=field_name,
+                value=value
+            )
+
+
+def validate_list_of_non_negative_floats(
+        field_name,
+        value,
+        min_length=1,
+        max_length=float('inf')
+):
+    validate_list(field_name, value, min_length, max_length)
+    for element in value:
+        if not is_number_positive(element, float) and float(element) != 0.0:
+            raise exceptions.FieldError(
+                f"{field_name} should be a list of floats equal or greater 0.0",
+                field=field_name,
+                value=element
+            )
+
+
+def validate_enum(field_name, value, enum_class):
+    try:
+        enum_class(value)
+    except ValueError:
+        raise exceptions.FieldError(
+            f"{field_name} has not valid value for"
+            f" {enum_class.__name__} enum class",
+            field=field_name,
+            value=value,
+        )
+
+
+def validate_task_type_with_meta_parameters(
+        field_name,
+        task_type,
+        meta_parameters,
+        task_type_class,
+        task_type_meta_parameters,
+):
+    try:
+        task_type_class(task_type)
+        task_class = task_type_meta_parameters[task_type]
+        task_class(meta_parameters)
+    except (ValueError, KeyError):
+        raise exceptions.FieldError(
+            f"'task_type' has not valid value for"
+            f" {task_type_class.__name__} enum class",
+            field=field_name,
+            value=(task_type, meta_parameters),
+        )
+
+
+def validate_correct_meta_parameters_class(meta_parameters_classes, value):
+    if value is not None:
+        error_counter = 0
+        for script_package in meta_parameters_classes:
+            try:
+                script_package(value)
+            except KeyError:
+                error_counter += 1
+            if error_counter == len(meta_parameters_classes):
```

Comment:
why? shouldn't it look more like:
```python
def validate_correct_meta_parameters_class(meta_parameters_classes, value):
    error_counter = 0
    for script_package in meta_parameters_classes:
        try:
            script_package(value)
        except KeyError:
            error_counter += 1
    if error_counter > 0:
        raise exceptions.FieldError(
            'meta_parameters should be an instance one of '
            f'{meta_parameters_classes} classes',
            field='meta_parameters',
            value=value,
        )
```
`None` check seems unnecessary.

Reply:
It has to be `error_counter == len(meta_parameters_classes)` because I can't choose at first which script_package should be validate, so I need to iterate one by one and if none of these classes will be correct for script_packages which I got then it should raise exception

Reply:
right - at least one will always raise... :+1: